### ISSUE TITLE
Don't 404 on `api/card/<id>/unpersist` if the card is not persisted

### ISSUE THREAD

**Issue by dpsutton** [metabase/metabase#44857]
we 404 on no card which is sensible. But we also 404 on the persistence record

`(api/let-404 [persisted-info (t2/select-one PersistedInfo :card_id card-id)]`

We should just 204 there if it isn't already persisted.

https://metaboat.slack.com/archives/C01LQQ2UW03/p1719529206708639

**Comment by iethree** [comment]
agreed. see: https://github.com/metabase/metabase/blob/b797ea4f036b1cf7fb05483e652c28c47558b15b/frontend/src/metabase/query_builder/components/view/sidebars/ModelCacheManagementSection/ModelCacheManagementSection.tsx#L49-L51
